packages/eth-providers/src/base-provider.ts

```diff
@@ -1420,70 +1392,32 @@ export abstract class BaseProvider extends AbstractProvider {
 
     result.timestamp = Math.floor((await this.queryStorage('timestamp.now', [], result.blockHash)).toNumber() / 1000);
 
-    result.wait = async (confirms?: number, timeout?: number) => {
+    result.wait = async (confirms?: number, timeout_ms?: number) => {
       if (confirms === null || confirms === undefined) {
         confirms = 1;
+      } else if (confirms < 0) {
+        throw new Error('invalid confirms value');
       }
-      if (timeout === null || timeout === undefined) {
-        timeout = 0;
+      if (timeout_ms === null || timeout_ms === undefined) {
+        timeout_ms = 0;
+      } else if (timeout_ms < 0) {
+        throw new Error('invalid timeout value');
       }
 
-      return new Promise((resolve, reject) => {
-        const cancelFuncs: Array<() => void> = [];
+      let wait$ = timeout_ms ? this.head$.pipe(
+        timeout({
+          first: timeout_ms,
+          with: () => throwError(() => logger.makeError('timeout exceeded', Logger.errors.TIMEOUT, { timeout: timeout_ms })),
+        })) : this.head$;
 
-        let done = false;
+      wait$ = wait$.pipe(first((head) => head.number.toNumber() - startBlock === confirms));
```

Comment:
```suggestion
      wait$ = wait$.pipe(first((head) => head.number.toNumber() - startBlock + 1 === confirms));
```
startBlock is where the tx is mined, so when current block === startBlock, it's 1 confirmation